### ISSUE TITLE
[bug] Duplicated underlying var declarations generated for protocol mocks with init requirements

### ISSUE THREAD

**Issue by pablocornejo** [uber/mockolo#224]
Hi, I've found a bug where when mocking a protocol that has multiple init requirements that contain a parameter with the same name, the generated underlying var declaration gets duplicated.

For example, generating a mock for:

```swift
protocol MyProtocol {
    init(param: Any)
    init(param: Any, anotherParam: String)
}
```

Results in:

```swift
class MyProtocolMock: MyProtocol {
        private var _param: Any!
    private var _anotherParam: String!
    private var _param: Any!
    public init() { }
    required init(param: Any) {
        self._param = param
    }
    required init(param: Any, anotherParam: String = "") {
        self._param = param
        self._anotherParam = anotherParam
    }


}
```

...which fails to compile: `error: invalid redeclaration of '_param'`

A similar thing happens if using different types for param:

```swift
protocol MyProtocol {
    init(param: Int)
    init(param: Any)
}
```

Results in:

```swift
class MyProtocolMock: MyProtocol {
        private var _param: Any!
    private var _param: Int!
    public init() { }
    required init(param: Any) {
        self._param = param
    }
    required init(param: Int = 0) {
        self._param = param
    }


}
```

...`error: invalid redeclaration of '_param'`

**Comment by fummicc1** [comment]
@pablocornejo 
Thank you for sharing!
If you have a solution, feel free to send a PR.😄
I will start to investigate this issue.

**Comment by fummicc1** [comment]
Thank you. I could reproduce this issue.

After my investigation, I would like to confirm with you about the expected behavior.

In my opinion, the followings are expected behaviors.

#### duplicated init param name

- input

```swift
/// @mockable
protocol MyProtocol {
    init(param: Int)
    init(param: Any)
}
```

Before thinking the solution, let me say this code is not common in general. We can avoid this case by using typealias.

```swift
/// @mockable
protocol MyProtocol {
    associatedtype T
    init(param: T)
}

class MyProtocolMock: MyProtocol {
        private var _param: T!
    init() { }
    required init(param: T) {
        self._param = param
    }

    typealias T = Any
}
```

This is why we don't have to take care of this problem for now. How do you think of it?


#### duplicated variables to be used in initializer

-  input

```swift
protocol MyProtocol {
    init(param: Any)
    init(param: Any, anotherParam: String)
}
```

- output

```swift
///
/// @Generated by Mockolo
///






class MyProtocolMock: MyProtocol {
    private var _anotherParam: String!
    private var _param: Any!
    init() { }
    required init(param: Any) {
        self._param = param
    }
    required init(param: Any, anotherParam: String = "") {
        self._param = param
        self._anotherParam = anotherParam
    }


}
```

I think the aim of initializer is to assign value to property, so we can combine conflicting properties with single property.

**Comment by fummicc1** [comment]
Hello, I have fixed this issue in #231 , so let me close this issue. please tell me if there is something wrong. :)
